decider-bcl2fastq/src/main/java/ca/on/oicr/pde/deciders/Bcl2fastqDecider.java

```diff
@@ -604,30 +613,72 @@ public void setConfig(Map<String, String> config) {
                             if (oicrRunCompleteTouchFile.exists()) {
                                 //run is complete
                             } else {
-                                laneErrors.add(String.format("Lane has not completed sequencing ([%s] is missing).", oicrRunCompleteTouchFile.getAbsolutePath()));
+                                errors.add(String.format("Lane has not completed sequencing ([%s] is missing).", oicrRunCompleteTouchFile.getAbsolutePath()));
                             }
                         } else {
-                            laneErrors.add(String.format("Lane run_dir = [%s] is not accessible or does not exist.", runDir.getAbsolutePath()));
+                            errors.add(String.format("Lane run_dir = [%s] is not accessible or does not exist.", runDir.getAbsolutePath()));
                         }
                     } else {
-                        laneErrors.add(String.format("Lane run_dir = [%s].", (runDirs == null ? "" : Joiner.on(",").join(runDirs))));
+                        errors.add(String.format("Lane run_dir = [%s].", (runDirs == null ? "" : Joiner.on(",").join(runDirs))));
                     }
                 }
             }
 
-            if (!laneErrors.isEmpty()) {
+            if (!errors.isEmpty()) {
                 invalidLanes.add(laneName);
                 log.warn("Lane = [{}] can not be processed due to the following reasons:\n"
                         + "{}\n"
                         + "Lane provenance: [{}]\n"
                         + "Sample provenance: [{}]",
-                        laneName, Joiner.on("\n").join(laneErrors), Joiner.on(";").join(lps), Joiner.on(";").join(sps));
+                        laneName, Joiner.on("\n").join(errors), Joiner.on(";").join(lps), Joiner.on(";").join(sps));
             }
         }
 
         //remove invalid lanes from lanes to analyze set
         Set<String> lanesToAnalyze = Sets.difference(candidateLanesToAnalyze, invalidLanes);
 
+        if (noLaneSplittingMode) {
+            Set<String> noLaneSplittingModeLanesToAnalyze = new HashSet<>();
+
+            // get all samples in lanesToAnalyze set
+            List<ProvenanceWithProvider<SampleProvenance>> samplesToAnalyze = new ArrayList<>();
+            for (String laneName : lanesToAnalyze) {
+                samplesToAnalyze.addAll(laneNameToSampleProvenance.get(laneName));
+            }
+
+            // group by samples by run and lane
+            Map<String, Map<String, List<String>>> samplesToAnalyzeGroupedByRun = samplesToAnalyze.stream().collect(
+                    Collectors.groupingBy(s -> {
```

Comment:
For all of these `x -> { return x.foo; }`, you can reduce them to `x -> x.foo` for readability.